### ISSUE TITLE
activesupport: Array#to_sentence returns #ActiveSupport::SafeBuffer instead of #String in some cases

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1950]
To reproduce:

```
gem install activesupport
ruby -ractive_support/core_ext/string -e 'p [ActiveSupport::SafeBuffer.new("one")].to_sentence.class'
```

Actual:

```
ActiveSupport::SafeBuffer
```

Expected:

```
String
```

**Comment by bjfish** [comment]
A minimal example:
```
% jt ruby -e 'class Foo < String; end; a = Foo.new("test"); p "#{a}".class'
Foo
% ruby -e 'class Foo < String; end; a = Foo.new("test"); p "#{a}".class' 
String
```

**Comment by norswap** [comment]
Indeed, the issue is that string interpolation must concatenate its bits and to do so, starts by dup'ing the first string. If that's a String subclass, then the whole thing becomes a String subclass. Fix incoming.

**Comment by norswap** [comment]
Fixed by db44482f40e443df41e41621fe4f222dcb67d914
